test/integration/controller_test.go

```diff
@@ -0,0 +1,151 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"testing"
+	"time"
+
+	"k8s.io/client-go/1.5/kubernetes/fake"
+	"k8s.io/kubernetes/pkg/api/v1"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1alpha1"
+	"github.com/kubernetes-incubator/service-catalog/pkg/brokerapi"
+	fakebrokerapi "github.com/kubernetes-incubator/service-catalog/pkg/brokerapi/fake"
+	"github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+	scinformers "github.com/kubernetes-incubator/service-catalog/pkg/client/informers"
+	informers "github.com/kubernetes-incubator/service-catalog/pkg/client/informers/servicecatalog/v1alpha1"
+	"github.com/kubernetes-incubator/service-catalog/pkg/controller/wip"
+	"github.com/kubernetes-incubator/service-catalog/pkg/registry/servicecatalog/server"
+)
+
+// TestController is a very basic test to start
+//
+// need etcd running
+// start a fresh apiserver for the controller to talk to
+func TestController(t *testing.T) {
+	fakeKubeClient, catalogClient, fakeBrokerCatalog, _, _, testController, _, stopCh := newTestController(t)
+	defer close(stopCh)
+
+	t.Log(fakeKubeClient, catalogClient, fakeBrokerCatalog, testController, stopCh)
+
+	fakeBrokerCatalog.RetCatalog = &brokerapi.Catalog{
+		Services: []*brokerapi.Service{
+			{
+				Name:        "test-service",
+				ID:          "12345",
+				Description: "a test service",
+				Plans: []brokerapi.ServicePlan{
+					{
+						Name:        "test-plan",
+						Free:        true,
+						ID:          "34567",
+						Description: "a test plan",
+					},
+				},
+			},
+		},
+	}
+	name := "test-name"
+	broker := &v1alpha1.Broker{
+		ObjectMeta: v1.ObjectMeta{Name: name},
+		Spec: v1alpha1.BrokerSpec{
+			URL:     "https://example.com",
+			OSBGUID: "OSBGUID field",
+		},
+	}
+	brokerClient := catalogClient.Servicecatalog().Brokers()
+
+	brokerServer, err := brokerClient.Create(broker)
+	if nil != err {
+		t.Fatalf("error creating the broker %q (%q)", broker, err)
+	}
+
+	timeout := time.After(30 * time.Second)
+	tick := time.Tick(500 * time.Millisecond)
+checkStatus:
+	for {
+		select {
+		case <-timeout:
+			t.Fatalf("timed out trying to observe a broker condition change")
+		// Got a tick, we should check on doSomething()
+		case <-tick:
+			brokerServer, err = brokerClient.Get(name)
+			if err != nil {
+				t.Fatalf("error getting broker %s (%s)", name, err)
+			}
+			t.Log(brokerServer)
+			t.Log(brokerServer.Status)
+			if len(brokerServer.Status.Conditions) > 0 {
+				break checkStatus
+			}
+		}
+	}
+
+	// cleanup our broker
+	err = brokerClient.Delete(name, &v1.DeleteOptions{})
+	if nil != err {
+		t.Fatalf("broker should be deleted (%s)", err)
+	}
+}
+
+func newTestController(t *testing.T) (
```

Comment:
copied and adjusted from https://github.com/kubernetes-incubator/service-catalog/blob/master/pkg/controller/wip/controller_test.go

uses the framework to generate server and associated client.

candidate for refactor into framework.